tsl/test/expected/continuous_aggs_drop_chunks.out

```diff
@@ -139,3 +139,139 @@ SELECT drop_chunks('records', '2000-03-16'::timestamptz);
  _timescaledb_internal._hyper_1_15_chunk
 (15 rows)
 
+\set VERBOSITY terse
+DROP MATERIALIZED VIEW records_monthly;
+NOTICE:  drop cascades to 32 other objects
+DROP TABLE records;
+DROP TABLE clients;
+\set VERBOSITY default
+-- Test how caggs are affected by drop_chunk due to refresh on drop.
+-- Demonstrates fix of issue #2592.
+CREATE OR REPLACE FUNCTION test_int_now() returns INT LANGUAGE SQL STABLE as 
+    $$ SELECT 125 $$;
+CREATE TABLE conditions(time_int INT NOT NULL, device INT, value FLOAT);
+SELECT create_hypertable('conditions', 'time_int', chunk_time_interval => 10);
+    create_hypertable    
+-------------------------
+ (3,public,conditions,t)
+(1 row)
+
+INSERT INTO conditions
+SELECT time_val, time_val % 4, 3.14 FROM generate_series(0,100,1) AS time_val;
+SELECT set_integer_now_func('conditions', 'test_int_now');
+ set_integer_now_func 
+----------------------
+ 
+(1 row)
+
+CREATE MATERIALIZED VIEW conditions_7
+    WITH (timescaledb.continuous, timescaledb.materialized_only = TRUE)
+    AS
+        SELECT time_bucket(7, time_int) as bucket,
+            SUM(value), COUNT(value)
+        FROM conditions GROUP BY bucket WITH DATA;
+NOTICE:  refreshing continuous aggregate "conditions_7"
+HINT:  Use WITH NO DATA if you do not want to refresh the continuous aggregate on creation.
+SELECT materialization_hypertable_schema||'.'||materialization_hypertable_name AS mat_hyper
+FROM timescaledb_information.continuous_aggregates
+WHERE view_name = 'conditions_7' \gset
+CREATE VIEW see_cagg AS SELECT * FROM conditions_7 WHERE bucket < 70 ORDER BY bucket;
+SELECT * FROM see_cagg;
+ bucket |  sum  | count 
+--------+-------+-------
+      0 | 21.98 |     7
+      7 | 21.98 |     7
+     14 | 21.98 |     7
+     21 | 21.98 |     7
+     28 | 21.98 |     7
+     35 | 21.98 |     7
+     42 | 21.98 |     7
+     49 | 21.98 |     7
+     56 | 21.98 |     7
+     63 | 21.98 |     7
+(10 rows)
+
+-- This is the simplest case, when the updated bucket is inside a chunk, so it is expected
+-- that the update is always reflected after the refresh on drop.
+UPDATE conditions SET value = 4.00 WHERE time_int = 2;
+-- This case updates values in the bucket, which affects two different partials in
+-- two different chunks, which are dropped in the same call to drop_chunks.
+UPDATE conditions SET value = 4.00 WHERE time_int = 9;
+UPDATE conditions SET value = 4.00 WHERE time_int = 11;
+SELECT drop_chunks('conditions', 20);
+               drop_chunks               
+-----------------------------------------
+ _timescaledb_internal._hyper_3_65_chunk
+ _timescaledb_internal._hyper_3_66_chunk
+(2 rows)
+
+SELECT * FROM see_cagg;
+ bucket |  sum  | count 
+--------+-------+-------
+      0 | 22.84 |     7
+      7 |  23.7 |     7
```

Comment:
If you drop the two first chunks separately, I would expect bucket 7 to first refresh to 22.84 and then when the second chunk is dropped it would be 23.7. Why not test that case as I suggested?

Reply:
Ohh, I think this is tested below... sorry

Reply:
Your case is already tested and I commented two times that I am adding it. Please, check the code